cmd/nirvana/project/init.go

```diff
@@ -0,0 +1,782 @@
+	Function: message.ListMessages,
+	Parameters: []def.Parameter{
+		{
+			Source:      def.Query,
+			Name:        "count",
+			Default:     10,
+			Description: "Number of messages",
+		},
+	},
+	Results: def.DataErrorResults("A list of messages"),
+}
+
+var getMessage = def.Definition{
+	Method:   def.Get,
+	Summary: "Get Message",
+	Description: "Get a message by id",
+	Function: message.GetMessage,
+	Parameters: []def.Parameter{
+		def.PathParameterFor("message", "Message id"),
+	},
+	Results: def.DataErrorResults("A message"),
+}
+`
+}
+
+func (o *initOptions) templateMessage() string {
+	return `
+{{- if .Boilerplate -}}
+{{ .GoBoilerplate }}
+{{- end }}
+
+package message
+
+import (
+	"context"
+	"fmt"
+)
+
+// Message describes a message entry.
+type Message struct {
+	ID      int    ` + "`json:\"id\"`" + `
+	Title   string ` + "`json:\"title\"`" + `
+	Content string ` + "`json:\"content\"`" + `
+}
+
+// ListMessages returns all messages.
+func ListMessages(ctx context.Context, count int) ([]Message, error) {
+	messages := make([]Message, count)
+	for i := 0; i < count; i++ {
+		messages[i].ID = i
+		messages[i].Title = fmt.Sprintf("Example %d", i)
+		messages[i].Content = fmt.Sprintf("Content of example %d", i)
+	}
+	return messages, nil
+}
+
+// GetMessage return a message by id.
+func GetMessage(ctx context.Context, id int) (*Message, error) {
+	return &Message{
+		ID:      id,
+		Title:   "This is an example",
+		Content: "Example content",
+	}, nil
+}
+`
+}
+
+func (o *initOptions) templateVersion() string {
+	return `
+{{- if .Boilerplate -}}
+{{ .GoBoilerplate }}
+{{- end }}
+
+package version
+
+// Following values should be substituted with a real value during build.
+var (
+	Version = "Unknown"
+	Commit = "Unknown"
+	Package = "{{ .ProjectPackage }}"
+)
+`
+}
+
+func (o *initOptions) templateDockerfile() string {
+	return `
+{{- if .Boilerplate -}}
+{{ .SharpBoilerplate }}
+
+{{ end -}}
+FROM {{ .BuildImage }}
+
+WORKDIR /go/src/{{ .ProjectPackage }}
+
+COPY . .
+
+ENV GOPATH /go
+
+ARG CMD_DIR=./cmd
+
+ARG ROOT={{ .ProjectPackage }}
+
+ARG TARGET={{ .ProjectName }}
+
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64                      \
+	go build -i -v -o /tmp/${TARGET}                  \
+	-ldflags "-s -w -X ${ROOT}/pkg/version.Version=${VERSION}  \
+	-X ${ROOT}/pkg/version.Commit=${COMMIT}                    \
+	-X ${ROOT}/pkg/version.Package=${ROOT}"                    \
+	${CMD_DIR}/${TARGET};
+
+FROM {{ .RuntimeImage }}
+
+ARG TARGET={{ .ProjectName }}
+
+COPY --from=0 /tmp/${TARGET} /${TARGET}
+
+ENTRYPOINT [/${TARGET}]
+`
+}
+
+func (o *initOptions) templateGopkg() string {
+	return `
+{{- if .Boilerplate -}}
+{{ .SharpBoilerplate }}
+#
+{{ end -}}
+# Gopkg.toml example
+#
+# Refer to https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
```

Comment:
what's this .... do you need them?

Reply:
It's document for users. Of course, we can remove it.